### PR TITLE
Use exponential estimator for Heapster addon resizer

### PR DESCRIPTION
**Pull request by piosz**
cc @fgrzadkowski @mwielgus @roberthbailey 
[Analytics]()

### CODE REVIEW

**Comment by piosz** [comment]
Do not merge, there are some issues with it:
```
I0609 20:22:38.954902       1 nanny_lib.go:102] The expected resources are %!v(PANIC=reflect.Value.Interface: cannot return value obtained from unexported field or method)
```

**Comment by roberthbailey** [comment]
I think we need to update the gci scripts as well, since they don't use these salt-defaults.

@andyzheng0831

**Comment by Q-Lee** [comment]
We should add a comment to change both locations, since this would have been twice it slipped through.

**Comment by piosz** [comment]
I fixed the issue in https://github.com/kubernetes/contrib/pull/1181. I'll update PR.

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -721,14 +721,14 @@ function start-kube-addons {
...
+    base_metrics_memory="140Mi"
     metrics_memory="${base_metrics_memory}"
-    base_eventer_memory="200Mi"
+    base_eventer_memory="190Mi"
     eventer_memory="${base_eventer_memory}"
     local -r metrics_memory_per_node="4"
     local -r eventer_memory_per_node="500"
     if [[ -n "${NUM_NODES:-}" && "${NUM_NODES}" -ge 1 ]]; then
-      num_kube_nodes="$((${NUM_NODES}-1))"
+      num_kube_nodes="$((${NUM_NODES}+1))"
```
Comment (piosz) [inline cluster/gce/gci/configure-helper.sh]: Not sure why there is {NUM_NODES}-1? Do we want to count also master, so it should be +1 instead? Any ideas?

cc @andyzheng0831 @roberthbailey @Q-Lee

Comment (andyzheng0831) [inline cluster/gce/gci/configure-helper.sh]: When I made the code change for GCI config, I simply followed the logic in the corresponding salt config file (e.g., https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml). I assumed that logic there is correct, so I did not have to spend tons of time on understanding every piece of details. If you notice the logic in the salt-based config changes, please feel free to change GCI config. Thanks

**Comment by andyzheng0831** [comment]
The change to GCI and Trusty config lgtm. Leave it to other reviewers for further checking other parts.

**Comment by Q-Lee** [comment]
The Jenkins test failures look unrelated, so on the assumption your base memory's are justified, this lgtm.  I'll wait for mwielgus to add the label after you get a passing test run.

**Comment by mwielgus** [comment]
lgtm

**Comment by piosz** [comment]
@k8s-bot unit test this issue: #ignore

**Comment by piosz** [comment]
@k8s-bot e2e test this issue: #26834

**Pull request merged**